packages/bruno-cli/src/utils/axios-instance.js

```diff
@@ -0,0 +1,40 @@
+const axios = require('axios');
+
+/**
+ * Function that configures axios with timing interceptors
+ * Important to note here that the timings are not completely accurate.
+ * @see https://github.com/axios/axios/issues/695
+ * @returns {import('axios').AxiosStatic}
+ */
+function makeAxiosInstance() {
+  /** @type {import('axios').AxiosStatic} */
+  const instance = axios.create();
+
+  instance.interceptors.request.use((config) => {
+    config.headers['request-start-time'] = Date.now();
+    return config;
+  });
+
+  instance.interceptors.response.use(
+    (response) => {
+      const end = Date.now();
+      const start = response.config.headers['request-start-time'];
+      response.headers['request-duration'] = end - start;
+      return response;
+    },
+    (error) => {
+      if (error.response) {
+        const end = Date.now();
+        const start = error.config.headers['request-start-time'];
+        error.response.headers['request-duration'] = end - start;
+      }
```

Comment:
2cents: these 2 don't need to be inside the `if`, only the `error.response.headers['request-duration']` part.

It might be useful to be able to zero it if duration is not present etc
       

```
const end = Date.now();
const start = error.config.headers['request-start-time'];
```

Reply:
Not sure I understand, what do you want to zero?

Reply:
because of the `if` we won't get `headers['request-duration']` if error is not a response, meaning it will be `undefined` 

in the line where you display it (`${responseTime} ms`) it needs to either be `0` or not be displayed for example

Reply:
You can probably solve it by always returning the duration or just

```
// defaults to 0 if header is not present
responseTime = response.headers.get('request-duration') ?? 0
```

Reply:
ah gotcha 👍🏽 , then it's all good

Reply:
Thanks for the feedback :+1: